src/components/article-body/body-components/heading.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { ComponentFactory } from '../component-factory';
+import { ArticleNode } from '../../../interfaces/article-body-node';
+
+export const Heading = ({
+    depth,
+    children,
+    id,
+    ...rest
+}: {
+    depth: string;
+    children: any;
+    id: string;
+}) => {
+    const parentNode = {
+        type: 'heading',
+        depth: depth,
+    };
+    return (
+        <div id={id}>
+            <a href={`#${id}`} />
+            {children.map((child: ArticleNode, index: number) => (
+                <ComponentFactory
+                    nodeData={child}
+                    key={index}
```

Comment:
Don't use the index as the key as it could cause issues in React https://medium.com/geekculture/reactjs-why-index-as-a-key-is-an-anti-pattern-4b9dc6ef0067